src/cache-unzip/src/unzip.ts

```diff
@@ -143,15 +141,24 @@ export const main = async (
         chunks.push(unz)
         cache.set(key, Buffer.concat(chunks, headLength + unz.length))
       }
+      return true
     }),
   )
   await cache.promise()
-  // TS mistakenly things didSomething is always false
-  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-  if (!didSomething) process.exit(1)
+  if (!didSomething.some(Boolean)) process.exit(1)
 }
 
-if (process.argv[1] === import.meta.filename) {
+if (
+  /* c8 ignore next */
+  (process.env.__VLT_INTERNAL_MAIN ?? process.argv[1]) ===
+  __CODE_SPLIT_SCRIPT_NAME
+) {
   process.title = 'vlt-cache-unzip'
```

Comment:
consider:

```ts
/* c8 ignore next */
const file = process.env.__VLT_INTERNAL_MAIN ?? process.argv[1]
if (file === __CODE_SPLIT_NAME) {
```

Reply:
Even though it's an immediate throwaway var, it makes the "what is this noise" clearer, and ends up being fewer lines to look at.